server/src/test/java/org/elasticsearch/index/analysis/JsonAnalyzerTests.java

```diff
@@ -0,0 +1,87 @@
+package org.elasticsearch.index.analysis;
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import com.fasterxml.jackson.core.JsonParseException;
+import org.apache.lucene.analysis.Analyzer;
+import org.apache.lucene.analysis.BaseTokenStreamTestCase;
+import org.elasticsearch.index.analysis.JsonAnalyzer;
+import org.elasticsearch.test.ESTokenStreamTestCase;
+
+import java.io.IOException;
+
+public class JsonAnalyzerTests extends ESTokenStreamTestCase {
+
+    public void testTextValues() throws Exception {
+        assertAnalyzesTo(
+            "{ \"key1\": \"value1\", \"key2\": \"value2\" }",
+            new String[] { "value1", "value2" });
+    }
+
+    public void testNumericValues() throws Exception {
+        assertAnalyzesTo(
+            "{ \"key\": 2.718 }",
+            new String[] { "2.718" });
+    }
+
+    public void testBooleanValues() throws Exception {
+        assertAnalyzesTo(
+            "{ \"key\": false }",
+            new String[] { "false" });
+    }
+
+    public void testNullValues() throws Exception {
+        assertAnalyzesTo(
+            "{ \"key\": null }",
+            new String[] { "null" });
+    }
```

Comment:
Are we going to be able to tell the difference between a null key, and a string with the value `"null"`?  Do we need to?

Reply:
I will think about this as I move over to the new approach -- it would be good to have better `null` handling.